### PR TITLE
Giveaways module

### LINKED ISSUE #152: Giveaways module
**Feature: Browsing the currently available giveaways**
**Given** I am a regular user of the server
**When** I invoke the `/giveaway view` command
**Then** a message appears with the following parts
- Embed: displays a single available giveaway's details (title, expiration, type, claim URL)
- Combo box: displays up to 5 giveaways' titles that, when selected, changes the embed's content
- Paginator: pages between up to 5 giveaways in the combo box

**Feature: Announcing newly discovered giveaways**
**Given** I am a the bot
**When** I scrape new giveaways using the data scrapers
**Then** I send a message containing up to 5 titles to the project server's announcement channel

**Feature: Subscribing to giveaway announcements**
**Given** I am the administrator of the server
**When** I invoke the `/giveaway announce channel: <channel>` command
**Then** the bot subscribes <channel> to the giveaway announcement channel of the project server

**Feature: Unsubscribing from giveaway announcements**
**Given** I am the administrator of the server
**When** I invoke the `/giveaway announce` command
**Then** the bot unsubscribes all channels of the server from the giveaway announcement channel of the project server

### PR DESCRIPTION
**Pull request by rexor12**
Closes #152

### CODE REVIEW

File: holobot/extensions/giveaways/models/giveaway_settings.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from dataclasses import dataclass
```
Comment (rexor12) [inline holobot/extensions/giveaways/models/giveaway_settings.py:1]: Delete unused file.

File: .vscode/holobot.code-snippets [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+	// Place your snippets for python here. Each snippet is defined under a snippet name and has a prefix, body and
```
Comment (rexor12) [inline .vscode/holobot.code-snippets:2]: Delete comment.

File: holobot/framework/i18n/i18n_provider.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import glob
 import os
 from json import load
-from typing import Any
+from typing import Any, Sequence
```
Comment (rexor12) [inline holobot/framework/i18n/i18n_provider.py:4]: Import Sequence from collections.abc.

File: holobot/sdk/database/queries/exists_builder.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from typing import Any, Tuple
```
Comment (rexor12) [inline holobot/sdk/database/queries/exists_builder.py:1]: Use `tuple`.

File: holobot/sdk/utils/datetime_utils.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from datetime import datetime, tzinfo
+from typing import Optional
```
Comment (rexor12) [inline holobot/sdk/utils/datetime_utils.py:2]: Use `|` instead of `Optional`.

File: holobot/sdk/utils/timedelta_utils.py [hunk 6]
```diff
@@ -1,23 +1,44 @@
 from datetime import timedelta
+from typing import List, Optional, Sequence, Tuple
```
Comment (rexor12) [inline holobot/sdk/utils/timedelta_utils.py:2]: Use `list`, `|`, `tuple` instead; import `Sequence` from `collections.abc`.

**Review by tooruu — APPROVED** [review]

**Comment by tooruu** [comment]
LGTM!

File: holobot/extensions/giveaways/scraper_runner.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        self.__process_delay = self.__configurator.get("Giveaways", "RunnerDelay", DEFAULT_DELAY)
+        self.__token_source: CancellationTokenSource | None = None
+        self.__background_task: Awaitable[None] | None = None
+
+    async def start(self) -> None:
+        if not self.__configurator.get("Giveaways", "EnableScrapers", True):
+            self.__logger.info("Giveaway scraping is disabled by configuration")
+            return
+
+        if len(self.__scrapers) == 0:
```
[the remark this answers is not in the document]
Comment (tooruu) [reply holobot/extensions/giveaways/scraper_runner.py:49]: More occurrence of this code smell are present in the code, pls search :)

**Pull request merged**